src/types/package.go

```diff
@@ -21,7 +21,7 @@ type ZarfPackage struct {
 	Kind       ZarfPackageKind                 `json:"kind" jsonschema:"description=The kind of Zarf package,enum=ZarfInitConfig,enum=ZarfPackageConfig,default=ZarfPackageConfig"`
 	Metadata   ZarfMetadata                    `json:"metadata,omitempty" jsonschema:"description=Package metadata"`
 	Build      ZarfBuildData                   `json:"build,omitempty" jsonschema:"description=Zarf-generated package build data"`
-	Components []ZarfComponent                 `json:"components" jsonschema:"description=List of components to deploy in this package"`
+	Components []ZarfComponent                 `json:"components" jsonschema:"description=List of components to deploy in this package,minItems=1"`
```

Comment:
TIL about `minItems` for list validation in json schemas. Nice! This means we can delete [this code](https://github.com/defenseunicorns/zarf/blob/b72d4c8a5ed34b10b481cd3fe6c20337507ac064/src/types/validate.go#L52-L54)

https://json-schema.org/understanding-json-schema/reference/array#length

Reply:
We can't delete it yet as this PR does not introduce schema validation during create. But it is deleted in #2585